app/controllers/projects_controller.rb

```diff
@@ -7,19 +7,19 @@ def index
 
   def show
     @project = Project.find_by_id(params[:id])
-    if session[:user_id].nil?
+    if current_user.nil?
```

Comment:
How about we make another helper method for this:

```
helper_method :current_user, :logged_in?

def logged_in?
    session[:user_id]
end

def current_user
    @current_user ||= User.find(session[:user_id]) if logged_in?
end
```

So we can use `if logged_in?` to remove that logic from the controller. I think that would be more clear too.